topi/python/topi/math.py

```diff
@@ -206,3 +207,82 @@ def cast(x, dtype):
         The result.
     """
     return tvm.compute(x.shape, lambda *i: x(*i).astype(dtype))
+
+
+@tvm.target.generic_func
+def matmul(a, b, transpose_a, transpose_b):
```

Comment:
Since we already have transpose and dense. Maybe we should not support This. Note that a typical matrix multiplication schedule works well for certain layout but will do explicit transpose for others

Reply:
I agree with removing the tranpose_a, tranpose_b parameter.
I think it's better to keep matmul here because:
1. numpy is introducing matmul operator
2. Currently dense operator only support rhs with <=2 dimensions
3. matmul is building block for dense gradient

Reply:
the only problem is that matmul requires non-trivial scheduling, which is not currently here, this means topi support for this is not yet complete. Maybe we could consider a separate PR for this, and include a reasonable schedule

Reply:
I do have the plan to put gemm scheduling from [tvm tutorial](http://docs.tvmlang.org/tutorials/optimize/opt_gemm.html).

I will submit another PR for the scheduling, thanks

Reply:
Ok, can we remove matmul from this PR so that it is included in the next PR that comes with the schedule?

Reply:
sounds good!